lisa/analysis/pixel6.py

```diff
@@ -0,0 +1,100 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2015, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pandas as pd
+import holoviews as hv
+from holoviews import opts
+
+from lisa.datautils import df_refit_index
+from lisa.analysis.base import TraceAnalysisBase
+from lisa.trace import requires_events
+from lisa.notebook import plot_signal
+
+P6_EMETER_CHAN_NAMES = {
+    'S4M_VDD_CPUCL0': 'little',
+    'S3M_VDD_CPUCL1': 'mid',
+    'S2M_VDD_CPUCL2': 'big',
+}
+
+class Pixel6Analysis(TraceAnalysisBase):
+    """
+    Support for Pixel 6-specific analysis data
+
+    :param trace: input Trace object
+    :type trace: :class:`trace.Trace`
+    """
+
+    name = 'pixel6'
+
+###############################################################################
+# DataFrame Getter Methods
+###############################################################################
+    @TraceAnalysisBase.cache
+    @requires_events('pixel6_emeter')
+    def df_power_meter(self):
+        """
+        Get the power meter readings across the trace.
+
+        :retuns: A :class:`pandas.DataFrame` with:
+            * A ``little`` column (readings from the little cluster)
+            * A ``mid`` column (readings from the mid cluster)
+            * A ``big`` column (readings from the big cluster)
+        """
+        df = self.trace.df_event('pixel6_emeter')
+        df = df.pivot_table(values='value', index='ts', columns='chan_name').reset_index().rename(columns=P6_EMETER_CHAN_NAMES)
+        channels = list(P6_EMETER_CHAN_NAMES.values())
+        df['ts_diff'] = df['ts'].diff()
+        # convert channels to power usage delta in mW
+        for channel in channels:
+            df[channel] = df[channel].diff() / df["ts_diff"]
+        # convert ts to seconds
+        df['Time'] = df['ts'] / 1000
```

Comment:
that should already be the case, do you have a snippet of the input dataframe ? `Time` is normally the index and timestamps come from ftrace itself.

Reply:
There's something a bit strange happening there. the 'ts' field is included with the event so I was using this for consistency as the power usage is computed based on the deltas in that column.

```
                 __comm  __pid  __cpu       ts  chan      value       chan_name
Time
297.103991  kworker/6:2     85      6  1917075     0    4464332    S10M_VDD_TPU
297.103993  kworker/6:2     85      6  1917075     1   25916055  VSYS_PWR_MODEM
297.103993  kworker/6:2     85      6  1917075     2       9916   VSYS_PWR_RFFE
297.103994  kworker/6:2     85      6  1917075     3   35633092  S2M_VDD_CPUCL2
297.103995  kworker/6:2     85      6  1917075     4   25287151  S3M_VDD_CPUCL1
...                 ...    ...    ...      ...   ...        ...             ...
472.252146  kworker/6:2     85      6  2092223     3  381290070  S2M_VDD_CPUCL2
472.252146  kworker/6:2     85      6  2092223     4  101833620  S3M_VDD_CPUCL1
472.252146  kworker/6:2     85      6  2092223     5  150807532  S4M_VDD_CPUCL0
472.252147  kworker/6:2     85      6  2092223     6   15784792     S5M_VDD_INT
472.252147  kworker/6:2     85      6  2092223     7   76808208     S1M_VDD_MIF
```

What should we do with this?